File: clients/printnanny-octoprint/rust/src/printer_status.rs
// PrinterStatus represents a PrinterStatus model.
#[derive(Clone, Copy, Debug, Deserialize, Eq, Hash, Ord, PartialEq, PartialOrd, Serialize)]
pub enum PrinterStatus {
    #[serde(rename="CLOSED_WITH_ERROR")]
    ClosedWithError,
    #[serde(rename="CLOSED")]
    Closed,
    #[serde(rename="CONNECTING")]
    Connecting,
    #[serde(rename="DETECT_BAUDRATE")]
    DetectBaudrate,
    #[serde(rename="DETECT_SERIAL")]
    DetectSerial,
    #[serde(rename="ERROR")]
    Error,
    #[serde(rename="NONE")]
    None,
    #[serde(rename="OFFLINE")]
    Offline,
    #[serde(rename="OPEN_SERIAL")]
    OpenSerial,
    #[serde(rename="OPERATIONAL")]
    Operational,
    #[serde(rename="PAUSED")]
    Paused,
    #[serde(rename="PRINTING")]
    Printing,
    #[serde(rename="STARTING")]
    Starting,
    #[serde(rename="TRANSFERING_FILE")]
    TransferingFile,
    #[serde(rename="UNKNOWN")]
    Unknown,
}
impl Default for PrinterStatus {
    fn default() -> PrinterStatus {
        PrinterStatus::ClosedWithError
    }
}

File: clients/printnanny-octoprint/typescript/src/PrinterStatus.ts

enum PrinterStatus {
  CLOSED_WITH_ERROR = "CLOSED_WITH_ERROR",
  RESERVED_CLOSED = "CLOSED",
  CONNECTING = "CONNECTING",
  DETECT_BAUDRATE = "DETECT_BAUDRATE",
  DETECT_SERIAL = "DETECT_SERIAL",
  ERROR = "ERROR",
  NONE = "NONE",
  OFFLINE = "OFFLINE",
  OPEN_SERIAL = "OPEN_SERIAL",
  OPERATIONAL = "OPERATIONAL",
  PAUSED = "PAUSED",
  PRINTING = "PRINTING",
  TRANSFERING_FILE = "TRANSFERING_FILE",
  UNKNOWN = "UNKNOWN",
}
export { PrinterStatus };